code/game/mobs/living/living_defense.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+const {to_chat, has_component} = require('bluespess');
+const {random_zone} = require('./carbon/body_parts/helpers.js');
+
+module.exports = {
+	//Args were rearranged from TG's to be in the same order as the code
+	run_armor_check(def_zone = null, attack_flag = "melee", armour_penetration, penetrated_text = null, absorb_text = null, soften_text = null) {
```

Comment:
You really should make the last 4 named arguments, like this:

`, {armour_penetration, penetrated_text = null, absorb_text = null, soften_text = null} = {}`

Yes, equals in a {}. 

You'd call it like this: `run_armor_check(def_zone = whatever, attack_flag = "melee", {/* whatever the named arguments are*/});

And here you use : instead. Don't ask.